app/services/notification_emailer.rb

```diff
@@ -100,7 +100,13 @@ def perform_enqueue(type, delay)
       user = notification.user
       return unless user.active? || user.staged?
       return if SiteSetting.must_approve_users? && !user.approved? && !user.staged?
-      return if user.staged? && (type == :user_linked || type == :user_quoted)
+      if user.staged? &&
+           (
+             type == :user_linked || type == :user_quoted || type == :user_mentioned ||
+               type == :group_mentioned
```

Comment:
I considered doing this check a bit higher along the call stack, e.g. at `notification.rb`, but since https://meta.discourse.org/t/staged-users-getting-user-linked-emails/145202 was also handled here, I decided to consolidate it here.